phac-nml/irida-next
# Update: Concatenation Service

## What does this PR do and why?
_Describe in detail what your merge request does and why._

This PR updates the attachments concatenation service to work with `fq` files, as well as, updates the concatenated paired-end files names to follow the format of illumina paired-end files (`basename_S1_L001_R1_001`, `basename_S1_L001_R2_001`) if the type is `illumina_pe` otherwise default to `basename_1` and `basename_2`

## Screenshots or screen recordings
_Screenshots are required for UI changes, and strongly recommended for all other pull requests._

![image](https://github.com/phac-nml/irida-next/assets/25466211/b9b2f640-c2b8-45a1-8dac-a49cd0d83aca)


## How to set up and validate locally
_Numbered steps to set up and validate the change are strongly suggested._

1) Sign in as a user
2) Create a project and add a sample
3) Upload single-end fastq files with the extension `fq`
4) Open a rails console
5) Set a variable for the user in step 1
6) Set a variable for Sample.last
7) Get the attachment ids using the variable from step 6..for example `sample.attachments`
8) Run `Attachments::ConcatenationService.new(user, sample, {basename: 'new-file', attachment_ids: [ATTACHMENT_IDS]}).execute`
9) The concatenation should complete successfully
10) Try the above steps for single-end `fq.gz` files, and also for paired-end `fq` and `fq.gz` files. The concatenation should complete successfully
11) In the ui, verify the correct filenames are listed (for paired-end, illumina paired-end, single-end files), and the types are correct.

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.
